skopt/parameter.py

```diff
@@ -38,178 +37,156 @@ def inverse_transform(self, values):
         return 10**np.asarray(values)
 
 
-class Log(TransformerMixin):
-    """Natural logarithm transform."""
-    def fit(self, values):
-        return self
+class Distribution:
 
-    def transform(self, values):
-        return np.log(values)
+    @abc.abstractmethod
```

Comment:
The difference is that making it an `abc` will prevent the instance being created if it does not implement those methods, while raising a `NotImplementedError` will allow the instance being created.

(I am not a purist and I don't mind removing them)